src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -19,13 +19,14 @@ namespace System.ServiceProcess
     /// and manipulate it or get information about it.
     public class ServiceController : Component
     {
-        private string _machineName;
+        private string _machineName = DefaultMachineName; // Never null, always a valid name
```

Comment:
`var n = new ServiceController().DisplayName;` would throw NullReferenceException before this change, but succeeds after this change. Do we need a test for this?

Reply:
Actually it throws IOE now, but I'll add a test.